cmd/cost/list.go

```diff
@@ -259,8 +275,31 @@ func (o OUCost) getSum() (sum decimal.Decimal, unit string, err error) {
 	return
 }
 
+func (o *OUCost) getAccountClaims(kubeCli client.Client) {
+	for i, cost := range o.Costs {
+		accountclaim, err := accountget.GetAccountClaimFromAccountID(context.TODO(), o.options.kubeCli, cost.AccountID, common.AWSAccountNamespace)
+		if err == nil {
+			o.Costs[i].accountClaimName = accountclaim.Name
+			o.Costs[i].accountClaimNamespace = accountclaim.Namespace
+			continue
+		}
+		if strings.Contains(fmt.Sprint(err), "AccountClaim matched for Account CR") {
```

Comment:
why not just have the underlying error string be "AccountClaim CR not found" or "Account CR not found" (the string you want displayed) so all you need is line 294?

Also side note, is this a common Go pattern for handling different error types? Im coming from a Java background where we would would generally define a custom exception (e.g. `class AccountClaimCRNotFound extends Exception` and `class AccountCRNotFound extends Exception`) so you can get strong type checks.

Reply:
A error class hierarchy is not very common in go, and often you'll find yourself figure out the error type by string comparison as a quick solution. it feels hacky, but implementing better tooling often is a lot of effort. The k8s packages have a methodology for error checking where you can can call e.g. `isNotFound` to figure out if an error returned by the API servier is a 404.